radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyLedgerEntryStore.java

```diff
@@ -524,13 +527,26 @@ private static OptionalLong headerKeyEpoch(DatabaseEntry entry) {
 		) {
 			this.dbTxn = dbTxn;
 			this.db = db;
-			this.key = entry(indexableBytes);
 			this.indexableBytes = indexableBytes;
+			this.reverse = indexableBytes[0] == SubstateTypeId.VALIDATOR_STAKE_DATA.id();
 		}
 
 		private void open() {
 			this.cursor = db.openCursor(dbTxn, null);
-			this.status = cursor.getSearchKeyRange(key, substateIdBytes, value, null);
+			if (reverse) {
+				var copy = Arrays.copyOf(indexableBytes, indexableBytes.length);
+				var lastIndex = indexableBytes.length - 1;
+				if (copy[lastIndex] == (byte) (0xff)) {
+					throw new IllegalStateException("Unexpected last byte.");
+				}
+				copy[lastIndex]++;
```

Comment:
This doesn't look right to me.
- If the last byte is `0xff`, we should extend the indexable byte by 1 more byte
- What if there are duplicates? (although it's unlikely for validator stake data)

Reply:
This is why the check at line 539 doesnt allow `0xff`.